docs: grammar style guide (#1827)
asyncapi/community: docs/styleguide/grammar.md

---
title: Grammar
description: The style guide gives advice on how to implement grammar when creating tutorials and other forms of content for AsyncAPI.
weight: 100
---
# AsyncAPI Style Guide: Grammar

## Abbreviations and acronyms

When using abbreviations and acronyms, write the full word with the acronym or abbreviation in parenthesis. After that, use the abbreviation and acronym for them. Here's an example:

> "An application can be a micro-service, IoT (Internet of things) device (for example, a sensor), mainframe process, and more.

Author: ```suggestion
"An application can be a microservice, IoT (Internet of things) device (for example, a sensor), mainframe process, and more.
```

>  An application can be a micro-service, IoT (Internet of things) device (for example, a sensor), mainframe process, and more.
>  That aspect is particularly beneficial in setups like IoT, where topics are often assigned per device or device segment.

Reviewer: ```suggestion
"An application can be a micro-service, IoT (Internet of things) device (for example, a sensor), mainframe process, and more.
 That aspect is particularly beneficial in setups like IoT, where topics are often assigned per device or device segment."
```

 
 By using acronyms and abbreviations effectively, users would easily understand the context and meaning of the abbreviation or acronym and how it relates to the tutorial or guide.

 ## Active Voice 

 When writing documentation and other forms of content for AsyncAPI, use active voice whenever possible. Here's an example:

✅ **DO:** "In the context of channel addresses within AsyncAPI documents, parameters play a crucial role in defining the dynamic components of an address."  

> ❌ **DON'T:** "In the context of channel addresses within AsyncAPI documents, parameters play a crucial role in defining the dynamic components of an address."

Reviewer: The two examples are exactly the same. Please iterate here?

> ❌ **DON'T:** "In the context of channel addresses within AsyncAPI documents, parameters play a crucial role in defining the dynamic components of an address."

Author: ```suggestion
❌ **DON'T:** "In the context of channel addresses within AsyncAPI documents, parameters played a crucial role in defining the dynamic components of an address."
```

- Use "color" instead of "colour"
- Use "center" instead of "centre".
- Use "organization" instead of "organisation". 
- Use "program" instead of "programme"

## Verb Tense
When writing documentation and other forms of content for AsyncAPI, use the present tense. Here's an example:
✅ **DO:** "The dynamic segment of each channel address, which corresponds to the device identifier, is then articulated through the use of parameters."
❌ **DON'T:** "The dynamic segment of each channel address, which corresponds to the device identifier, was then articulated through the use of parameters."

## Personal Pronouns

 Use personal pronouns sparingly when writing tutorials and other forms of content for AsyncAPI to make them more human and approachable to users.  Here's an example:

 ✅ **DO:** "You can add parameters to the `channel.address` by adding a parameter between curly braces like {`braces`}. Next, use `channel.parameters` to define your parameters. Finally, leverage the `components.parameters` to enable reusable parameters' definitions across multiple channels."

>  ❌ **DON'T:** "You can add parameters to the `channel.address` by adding a parameter between curly braces like {`braces`}. Next, use `channel.parameters` to define your parameters. Finally, you can leverage the `components.parameters` to enable reusable parameters' definitions across multiple channels."

Author: ```suggestion
 ❌ **DON'T:** "You can add parameters to the `channel.address` by adding a parameter between curly braces like {`braces`}. Next, use `channel.parameters` to define your parameters. Finally, leverage the `components.parameters` to enable reusable parameters' definitions across multiple channels."
```

 ## Additional Resources 

 If you're looking for more information on grammar and style, consider checking out the following guides:

 - [Google Style Guide](https://developers.google.com/style)

>  - [Microsoft Style Guide](https://developers.google.com/style)

Author: ```suggestion
 - [Microsoft Style Guide](https://learn.microsoft.com/en-us/style-guide/welcome/)
```

---
title: Grammar
description: The style guide gives advice on how to implement grammar when creating tutorials and other forms of content for AsyncAPI.
weight: 100
---

> # AsyncAPI Style Guide: Grammar

Reviewer 2: ```suggestion
```

✅ **DO:** "In the context of channel addresses within AsyncAPI documents, parameters play a crucial role in defining the dynamic components of an address."  
❌ **DON'T:** "In the context of channel addresses within AsyncAPI documents, the dynamic components of an address were defined by parameters."

As shown in the above sentence, active voice makes the concept more engaging and easier to understand.

## Capitalization

When capitalizing titles, headings, and certain words and terms, use the following rules:
- Capitalize the first word of a sentence.
- Capitalize the first word in a heading and title.
- Capitalize all:
  - Nouns
  - Pronouns
  - Adjectives
  - Verbs
  - Adverbs
  - Prepositions (in, on, at, by, for, with, to, from, of, about, through, between, among, under, over, after, before, during)
  - Some subordinating conjunctions (if, unless, while, until, since, as, because, that, which, who, whom, whose, what, where, when, how, why)
- Avoid using all capital letters for emphasizing a term or concept (use italics sparingly).
- Avoid capitalizing:
  - Spelled-out acronyms unless they are proper nouns (API, HTTP, URL)
  - Articles (a, an, the)

>   - Conjunctions (and, but, or, nor, for, so, yet) unless they are part of a title or heading ("Parameters in channel address")## Spelling

Reviewer 2: ```suggestion
  - Conjunctions (and, but, or, nor, for, so, yet) unless they are part of a title or heading ("Parameters in channel address")
  
 ## Spelling
```

> AsyncAPI's content and documentation follows American English spelling standards. Here are some preferred spelling of certain words:

Reviewer 2: I wouldn't assume we use American English. Personally, I use British English for all my docs because that's the English language I know. Rather, remove this sentence or reconstruct it @CBID2 to be inclusive for all

Author: Done @thulieblack

> AsyncAPI's content and documentation follows American English spelling standards. Here are some preferred spelling of certain words:

Author: ```suggestion
AsyncAPI's content and documentation follows English spelling standards. Here are some preferred spelling of certain words:
```

- Use "color" instead of "colour"
- Use "center" instead of "centre".
- Use "organization" instead of "organisation". 
- Use "program" instead of "programme"

## Verb Tense

> When writing documentation and other forms of content for AsyncAPI, use the present tense. Here's an example:

Reviewer 2: ```suggestion
When writing documentation and other forms of content for AsyncAPI, use the present tense. Here's an example:

```

✅ **DO:** "The dynamic segment of each channel address, which corresponds to the device identifier, is then articulated through the use of parameters."
❌ **DON'T:** "The dynamic segment of each channel address, which corresponds to the device identifier, was then articulated through the use of parameters."

## Personal Pronouns

>  Use personal pronouns sparingly when writing tutorials and other forms of content for AsyncAPI to make them more human and approachable to users. Here's an example: 

Reviewer 2: ```suggestion
 Use personal pronouns sparingly when writing tutorials and other forms of content for AsyncAPI to make them more human and approachable to users. Here's an example: 
 
```

When writing documentation and other forms of content for AsyncAPI, use the present tense. Here's an example:
✅ **DO:** "The dynamic segment of each channel address, which corresponds to the device identifier, is then articulated through the use of parameters."
❌ **DON'T:** "The dynamic segment of each channel address, which corresponds to the device identifier, was then articulated through the use of parameters."

## Personal Pronouns

 Use personal pronouns sparingly when writing tutorials and other forms of content for AsyncAPI to make them more human and approachable to users. Here's an example: 
 ✅ **DO:** "You can add parameters to the `channel.address` by adding a parameter between curly braces like {`braces`}. Next, use `channel.parameters` to define your parameters. Finally, leverage the `components.parameters` to enable reusable parameters' definitions across multiple channels."
 ❌ **DON'T:** "You can add parameters to the `channel.address` by adding a parameter between curly braces like {`braces`}. Next, use `channel.parameters` to define your parameters. Finally, leverage the `components.parameters` to enable reusable parameters' definitions across multiple channels."

> 

Author: How can I do that @coderabbitai?

---
title: Grammar
description: The style guide gives advice on how to implement grammar when creating tutorials and other forms of content for AsyncAPI.
weight: 100
---

## Abbreviations and acronyms

When using abbreviations and acronyms, write the full word with the acronym or abbreviation in parenthesis. After that, use the abbreviation and acronym for them. Here's an example:

> An application can be a microservice, IoT (Internet of things) device (for example, a sensor), mainframe process, and more.
> That aspect is particularly beneficial in setups like IoT, where topics are often assigned per device or device segment.
 

>  By using acronyms and abbreviations effectively, users would easily understand the context and meaning of the abbreviation or acronym and how it relates to the tutorial or guide.

Reviewer 2: @CBID2 this is a valid review

## Capitalization

When capitalizing titles, headings, and certain words and terms, use the following rules:
- Capitalize the first word of a sentence.
- Capitalize the first word in a heading and title.
- Capitalize all:
  - Nouns
  - Pronouns
  - Adjectives
  - Verbs
  - Adverbs
  - Prepositions (in, on, at, by, for, with, to, from, of, about, through, between, among, under, over, after, before, during)
  - Some subordinating conjunctions (if, unless, while, until, since, as, because, that, which, who, whom, whose, what, where, when, how, why)
- Avoid using all capital letters for emphasizing a term or concept (use italics sparingly).
- Avoid capitalizing:
  - Spelled-out acronyms unless they are proper nouns (API, HTTP, URL)
  - Articles (a, an, the)
  - Conjunctions (and, but, or, nor, for, so, yet) unless they are part of a title or heading ("Parameters in channel address")
  
 ## Spelling
Here are some preferred spellings of certain words:
- Use "color" instead of "colour"
- Use "center" instead of "centre".
- Use "organization" instead of "organisation". 
- Use "program" instead of "programme"

> 

Reviewer 2: same here

> 

Author: Hey @thulieblack I made this and the [other coderabbitai comment](https://github.com/asyncapi/community/pull/1827#discussion_r2048594798) that you mentioned.
